src/node/node.js

```diff
@@ -396,7 +396,7 @@ class Node extends Base {
             }
             this.log('connection established'.green, this.formatNode(peer.data.hostname, peer.data.port));
             this._sendNeighbors(ws);
-            return peer.markConnected().then(() => this._ready && this.opts.onPeerConnected(peer));
+            return this.pingPeer(peer).then(([peer, ping]) => peer.markConnected(ping).then(() => this._ready && this.opts.onPeerConnected(peer)));
```

Comment:
you can bring `pingPeer` function to `Peer` class and chain all calls in a flat list

```
peer.ping().then(() => peer.markConnected()).then(() => this._ready && this.opts.onPeerConnected(peer));

```